engine/src/ast/function_expr.rs

```diff
@@ -76,6 +76,9 @@ impl<'s> FunctionCallExpr<'s> {
                 FunctionCallArgExpr::Literal(literal) => literal.into(),
             })
         }
+        for opt_arg in &self.function.opt_args[values.len() - self.function.args.len()..] {
```

Comment:
Can't comment on the corresponding line, but I suppose `with_capacity` above should be changed now?

Better yet, instead of using `with_capacity` manually, combine both `for` loops using iterator methods - they would be less error-prone, while having the same effect (they automatically take care of capacity calculation, so you don't need to maintain that manually in the code). It would look like:

```rust
let values: Vec<_> =
  self.args.iter()
  .map(|arg| ...)
  .chain(self.function.opt_args[...].iter().map(|opt_arg| ...)
  .collect();

Reply:
Done